src/lib/arch/aarch64/Instruction_execute.cc

```diff
@@ -4453,65 +5737,104 @@ void Instruction::execute() {
         const uint32_t sliceNum =
             (ws + metadata_.operands[0].sme_index.disp) % partition_num;
 
-        const uint64_t* tileSlice =
-            sourceValues_[sliceNum].getAsVector<uint64_t>();
-        memoryData_ = sve_merge_store_data<uint64_t>(tileSlice, pg, VL_bits);
+        std::vector<uint64_t> memData;
```

Comment:
Have you profiled the use of std::vector throughout. It might not make too much difference but "the internet" is saying they are always initialised (which might not be desirable for performance). Do you NEED 0 initialised memory? Do you need dynamically sized arrays? Or could this be done with a std::array or normal array? CPP reference also says reallocations (when resizing) could be costly, `reserve()` could be useful it if is a problem

Reply:
Given the vector length is unknown before simulation, and this instruction is predicated, to me the sensible and reasonable choice is a dynamic data structure such as a vector.